unicon_backend/evaluator/tasks/programming/steps.py

```diff
@@ -291,6 +329,29 @@ def run(self, var_inputs: dict[SocketName, ProgramVariable], *_) -> Program:
         ]
 
 
+class ObjectAccessStep(Step):
+    """
+    A step to retrieve a value from a dictionary.
+    To use this step, the user must provide the key value to access the dictionary.
+
+    Socket Name Format:
+    - DATA: for the dictionary
+    """
+
+    subgraph_socket_ids: ClassVar[set[str]] = set()
+    key: str
+
+    def run(self, var_inputs: dict[SocketName, ProgramVariable], *_) -> Program:
+        output_socket_name: str = self.outputs[0].id
+
+        # Assumption: input socket id is DATA
+        input_value = var_inputs["DATA"]
```

Comment:
are we able to assume that there must a single `DATA` type socket in a `ObjectAccessStep` node (enforced through `model_validator`)? This is so that we don't have to explicitly assume the socket name must always be `DATA` and we can have a arbitrary naming as long as it is `DATA.IN.X.X`

Reply:
22ab869